main.js

```diff
@@ -0,0 +1,45 @@
+const currencyData = [];
+
+function addCurrency(){
+
+    const baseCurrency = document.getElementById('baseCurrency').value;
+    const conjugateCurrency =document.getElementById('conjugateCurrency').value;
+    const rate = parseFloat(document.getElementById('rate').value);
+
+    const currentDate = new Date();
+    const timestamp = currentDate.getTime();
+    const formattedDate = currentDate.toISOString();
+
+    let newCurrency = {
+        base: baseCurrency,
+        conjugate: conjugateCurrency,
+        rate: rate,
+        timestamp: timestamp,
+        date: formattedDate
+    };
+
+    currencyData.push(newCurrency);
+    console.log(currencyData);
+
+    return currencyData;
+}
+
+
+function convertCurrency(){
+    let mainCurrency = document.getElementById('mainCurrency').value;
+    let targetCurrency = document.getElementById('targetCurrency').value;
+    let amount = parseFloat(document.getElementById('amount').value);
+
+    if (isNaN(amount)){
```

Comment:
I like how you're managing the error cases! Alternatively, you could set the `input` field as 'required'.

Reply:
Okay, that is true. I am going to add 'required' attribute to the input elements.
```
      <input type="text" value="DKK" id="baseCurrency" required/>
      <input type="text" id="conjugateCurrency" required/>
      <input type="number" id="rate" required/>
      <button type="button" onclick="addCurrency()">Add Currency</button>
    </form>
    <form id="convertionForm">
      <input type="text" value="DKK" id="mainCurrency" required/>
      <input type="text" id="targetCurrency" required/>
      <input type="number" id="amount" required/>
```